tmate-ssh-server.c

```diff
@@ -506,7 +518,50 @@ void tmate_ssh_server_main(struct tmate_session *session, const char *keys_dir,
 
 		ssh_bind_free(bind);
 
-		client_bootstrap(session);
+		client_bootstrap_cb(session,
+                             handle_role_daemon,
+                             handle_role_pty_client,
+                             handle_role_exec);
 		/* never reached */
 	}
 }
+
+void tmate_ssh_server_main(struct tmate_session *session, const char *keys_dir,
+                           const char *bind_addr, int daemon_port, int client_port)
+{
+	struct tmate_ssh_client *client = &session->ssh_client;
+	ssh_bind daemon_bind, client_bind;
+	pid_t pid;
+
+	tmate_catch_sigsegv();
+	signal(SIGCHLD, handle_sigchld);
+
+	daemon_bind = prepare_ssh(keys_dir, bind_addr, daemon_port);
+	client_bind = prepare_ssh(keys_dir, bind_addr, client_port);
+
+	client->session = ssh_new();
+	client->channel = NULL;
+	client->winsize_pty.ws_col = 80;
+	client->winsize_pty.ws_row = 24;
+	session->session_token = "init";
+
+	if (!client->session)
+		tmate_fatal("Cannot initialize session");
+
+    if ((pid = fork()) < 0)
+      tmate_fatal("Can't fork");
+
```

Comment:
minor: indent in the rest of the function (should be linux kernel style, with tabs).

Reply:
Fixed (I might have missed some spots).